src/IO/External/InterpolateFromCocal.cpp

```diff
@@ -0,0 +1,204 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "IO/External/InterpolateFromCocal.hpp"
+
+#include <array>
+#include <string>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+extern "C" {
+// Define the COCAL Fortran reader functions here for different id_types.
+void coc2cac_co(const int& npoints, const double* x, const double* y,
```

Comment:
since this is provided by COCAL itself, shouldn't there be a header file it provides that one can use to get this?

Reply:
there is not really much of a difference between a `&` and a `*`. Are you trying to imply that the `*` ones are actually arrays?

Reply:
a `const int &` make not much sense, I'd just pass by value. Unless you somehow are actually calling out to a Fortran routine, in which case this is all bad anyway since different Fortran compilers use different name mangling schemes.